raakinshabbir/recipeat---276
# Filter Page Pull Request

Overview:
This PR implements the filter page functionality for the recipe app. It allows users to filter recipes based on difficulty, cooking time, and unwanted ingredients. After applying filters, users are redirected to the swiping interface with the filtered recipes.

Changes Made:
Added a filter page with options to select:
-     Difficulty (Easy, Medium, Hard)
-     Maximum cooking time (in minutes)
-     Unwanted ingredients (comma-separated list)

- Integrated the filtering logic into the swiping interface so that only filtered recipes are shown.
- Modified ApplicationController to filter when needed.
- Modified the SwipeController to handle filtering.

Testing Instructions:
- Navigate to the filter page.
- Apply filters (e.g., select "Easy" for difficulty, set a max cooking time, and list unwanted ingredients (match unwanted ingredients exactly as ingredients enlisted in each of the recipes).
- After submitting the form will be moved to the swiping interface, check it shows only the recipes that match the filters.
- Verify that filters work correctly across different combinations of difficulty, cooking time, and unwanted ingredients.
